libs/agno/agno/agent/agent.py

```diff
@@ -669,7 +678,14 @@ def _run(
         reasoning_time_taken = 0.0
         if self.stream:
             model_response = ModelResponse()
-            for model_response_chunk in self.model.response_stream(messages=run_messages.messages):
+            for model_response_chunk in self.model.response_stream(
+                messages=run_messages.messages,
+                response_format=response_format,
+                tools=self._tools_for_model,
+                functions=self._functions_for_model,
+                tool_choice=self.tool_choice,
+                tool_call_limit=self.tool_call_limit,
```

Comment:
Thinking a general limit could be defined in the Model and be independent of the agent runs. Would be nice DX to set defaults when reusing a model. 
```python
model = OpenAIChat(..., tool_call_limit=5)
agent.run(..., tool_call_limit=10) # raises with InvalidLimit (or we use the min)
```

Not blocking, just an idea.

Reply:
That is a good idea!